holidays/countries/brunei.py

```diff
@@ -0,0 +1,447 @@
+
+    special_holidays = {
+        2017: ((OCT, 5, golden_jubilee_hassanal_bolkiah),),
+    }
+
+    def __init__(self, *args, **kwargs):
+        ChineseCalendarHolidays.__init__(self)
+        ChristianHolidays.__init__(self)
+        InternationalHolidays.__init__(self)
+        IslamicHolidays.__init__(self, calendar=BruneiIslamicCalendar())
+
+        super().__init__(*args, **kwargs)
+
+    def _populate(self, year):
+        # Available post-Independence from 1984 afterwards
+        if year <= 1983:
+            return None
+
+        def _add_observed(dt: date) -> None:
+            """
+            If Public Holiday falls on either Friday or Sunday, in-lieu
+            observance are given out on the following Saturday or Monday.
+            """
+            if self.observed and (self._is_friday(dt) or self._is_sunday(dt)):
+                for name in self.get_list(dt):
+                    self._add_holiday(
+                        self.tr("%s - Diperhatikan") % name, dt + td(days=+1)
+                    )
+
+        super()._populate(year)
+
+        # New Year's Day.
+        # Awal Tahun Masihi
+        # Status: In-Use.
+
+        _add_observed(self._add_new_years_day(tr("Awal Tahun Masihi")))
+
+        # Lunar New Year
+        # Tahun Baru Cina
+        # Status: In-Use.
+
+        _add_observed(self._add_chinese_new_years_day(tr("Tahun Baru Cina")))
+
+        # Isra Mi'raj
+        # Israk dan Mikraj
+        # Status: In-Use.
+
+        for dt in self._add_isra_and_miraj_day(tr("Israk dan Mikraj")):
+            _add_observed(dt)
+
+        # National Day
+        # Hari Kebangsaan
+        # Status: In-Use.
+        # Starts in 1984.
+
+        _add_observed(self._add_holiday(tr("Hari Kebangsaan"), FEB, 23))
+
+        # First Day of Ramadan
+        # Hari Pertama Berpuasa
+        # Status: In-Use.
+
+        for dt in self._add_ramadan_beginning_day(tr("Hari Pertama Berpuasa")):
+            _add_observed(dt)
+
+        # Anniversary of the revelation of the Quran
+        # Hari Nuzul Al-Quran
+        # Status: In-Use.
+
+        for dt in self._add_nuzul_al_quran_day(tr("Hari Nuzul Al-Quran")):
+            _add_observed(dt)
+
+        # Eid al-Fitr
+        # Hari Raya Aidil Fitri
+        # Status: In-Use.
+        # This is celebrate for three days in Brunei.
+        # Observed as 'Hari Raya Puasa' and only for 2 days at certain point.
+
+        name = tr("Hari Raya Aidil Fitri")
+
+        alfitr = self._add_eid_al_fitr_day(name)
+        self._add_eid_al_fitr_day_two(name)
+        self._add_eid_al_fitr_day_three(name)
+
+        # Eid al-Fitr (Observed)
+        # Hari Raya Aidil Fitri - Diperhatikan
+        # 1: If Wed-Thu-Fri -> Sat (+3)
+        # 2: If Thu-Fri-Sat -> Mon (+4)
+        # 3: If Fri-Sat-Sun -> Mon (+3)
+        # 4: If Sat-Sun-Mon -> Tue (+3)
+        # 5: If Sun-Mon-Tue -> Wed (+3)
+
+        aidil_fitri_in_lieu = self.tr("%s - Diperhatikan") % self.tr(
```

Comment:
This way,` Eid al-Fitr (Observed)` will never be labeled as "estimated", regardless of the origin of the date (exact or not). But unfortunately, I don't know of a good way to avoid it now. 🤷‍♂️

Reply:
@arkid15r any ideas on this?

Reply:
If I understand the problem correctly you're looking for something like:


```python
self._add_islamic_calendar_holiday(
    self.tr("%s - Diperhatikan") % self[hol_date],
    (
        (
            obs_date,
            self._year
            not in BruneiIslamicCalendar.EID_AL_FITR_DATES,
        ),
    ),
)
```

Reply:
Fixed.